lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -1280,6 +1280,112 @@ void BoundInterpreterFunction::fwdBatchOneHotInst(
   }
 }
 
+// Example:
+// Depth 0                    Depth 1
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+// 0 1 0 1 0 1 0 1 0 1        a b a b a b a b a b
+// 2 3 2 3 2 3 2 3 2 3        c d c d c d c d c d
+//
+// after transformation
+// blockDepth0
+//  depth 0          depth 1
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// 0 0 0 0 0        a a a a a
+// blockDepth 1
+//  depth 2          depth 3
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// 1 1 1 1 1        b b b b b
+// blockDepth 2
+// depth 4           depth 5
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// 2 2 2 2 2        c c c c c
+// blockDepth 3
+// depth 6           depth 7
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+// 3 3 3 3 3        d d d d d
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdSpaceToDepthInstImpl(
+    const glow::SpaceToDepthInst *I) {
+  auto *inT = getTensor(I->getSrc());
+  auto *outT = getTensor(I->getDest());
+
+  ElemTy *inPtr = reinterpret_cast<ElemTy *>(inT->getUnsafePtr());
```

Comment:
Let's not use `getUnsafePtr()`, because Interpreter backend is meant to be safe and understandable. Please do `auto inH = inT->getHandle<ElemTy>()`, and use `inH.at({b, ih, iw, ic})`